serde-support/src/main/java/io/micronaut/serde/support/util/JsonNodeDecoder.java

```diff
@@ -289,10 +288,7 @@ public String decodeKey() {
 
         @Override
         public void skipValue() {
-            if (complete) {
-                throw new IllegalStateException("Already drained");
-            }
-            complete = true;
+            // No-op: Allow to read multiple times
```

Comment:
ohhhhhh. no we should definitely not allow this. it will break with more efficient implementations.

Reply:
i will try a different solution

Reply:
Yeah, maybe we can allow to clone the buffer